lib/internal/util.js

```diff
@@ -366,6 +366,43 @@ function spliceOne(list, index) {
   list.pop();
 }
 
+let getStructuredStack;
+function isInsideNodeModules() {
+  if (!getStructuredStack) {
+    // Lazy-load to avoid a circular dependency.
+    const { runInNewContext } = require('vm');
+    // Use `runInNewContext()` to get something tamper-proof and
+    // side-effect-free. Since this is currently only used for a deprecated API,
+    // the perf implications should be okay.
+    getStructuredStack = runInNewContext('(' + function() {
+      Error.prepareStackTrace = function(err, trace) {
+        err.stack = trace;
+      };
+      Error.stackTraceLimit = Infinity;
```

Comment:
`Infinity`? Why? That would mean that a function that does `new Buffer` in a callback of e.g. `express` would not get warning, correct? Shouldn't this check just the location of the file that does `new Buffer()`?

Reply:
Ah, disregard this one comment, I think I understand how this works now, sorry =).
Not deleting just to be double-sure.